File: pet-hospital-miniprogram/src/api/consultation.ts
import request from './request'

export interface Consultation {
    id?: number
    customerId?: number
    petId?: number
    title: string
    content: string
    images?: string
    status?: string
    replyContent?: string
    replyDoctorId?: number
    replyTime?: string
    createdAt?: string
    updatedAt?: string
    pet?: any
    doctor?: any
}

export const consultationApi = {
    // 创建咨询
    create: (data: Consultation) => {
        return request.post('/api/consultations/create', data)
    },

    // 获取我的咨询列表
    getMyConsultations: (params?: { status?: string; page?: number; size?: number }) => {
        return request.get('/api/consultations/my-consultations', params)
    },

    // 获取咨询详情
    getDetail: (id: number) => {
        return request.get(`/api/consultations/${id}`)
    }
}

File: pet_hospital_backend/src/main/java/com/pethospital/entity/Consultation.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("consultations")
public class Consultation {
    
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long customerId;
    
    private Long petId;
    
    private String title;
    
    private String content;
    
    private String images; // JSON数组字符串
    
    private String status; // PENDING, ANSWERED, CLOSED
    
    private String replyContent;
    
    private Long replyDoctorId;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    private LocalDateTime replyTime;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss")
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
    
    
    private LocalDateTime deletedAt;
    
    // 关联对象 (非数据库字段)
    @TableField(exist = false)
    private Pet pet;
    
    @TableField(exist = false)
    private Employee doctor;
}
